packages/backend/src/services/InstanceConfigurationService/InstanceConfigurationService.ts

```diff
@@ -0,0 +1,290 @@
+
+    private readonly userModel: UserModel;
+
+    private readonly organizationAllowedEmailDomainsModel: OrganizationAllowedEmailDomainsModel;
+
+    private readonly personalAccessTokenModel: PersonalAccessTokenModel;
+
+    private readonly emailModel: EmailModel;
+
+    private readonly projectService: ProjectService;
+
+    private readonly serviceAccountModel?: ServiceAccountModel;
+
+    constructor({
+        lightdashConfig,
+        analytics,
+        organizationModel,
+        projectModel,
+        userModel,
+        organizationAllowedEmailDomainsModel,
+        personalAccessTokenModel,
+        emailModel,
+        projectService,
+        serviceAccountModel,
+    }: InstanceConfigurationServiceArguments) {
+        super();
+        this.lightdashConfig = lightdashConfig;
+        this.analytics = analytics;
+        this.organizationModel = organizationModel;
+        this.projectModel = projectModel;
+        this.userModel = userModel;
+        this.organizationAllowedEmailDomainsModel =
+            organizationAllowedEmailDomainsModel;
+        this.personalAccessTokenModel = personalAccessTokenModel;
+        this.emailModel = emailModel;
+        this.projectService = projectService;
+        this.serviceAccountModel = serviceAccountModel;
+    }
+
+    async initializeInstance() {
+        // No permissions check here, there are no users yet
+        // No initial setup, we skip this step
+        if (!this.lightdashConfig.initialSetup) {
+            this.logger.debug(`No initial setup config found, skipping`);
+            return;
+        }
+        try {
+            const setup = this.lightdashConfig.initialSetup;
+            // If no project is set, we can create a new one using environment variables
+            const hasOrgs = await this.organizationModel.hasOrgs();
+
+            if (hasOrgs) {
+                this.logger.debug(
+                    `Initial setup: There is already an organization, we skip this initial setup`,
+                );
+                // There is already an organization, we skip this step
+                return;
+            }
+            const hasAnyProjects = await this.projectModel.hasAnyProjects();
+
+            if (hasAnyProjects) {
+                // This should not happen, since we can't have projects without orgs
+                throw new UnexpectedServerError(
+                    `Initial setup: cannot initialize instance, there are already defined projects. Exiting`,
+                );
+            }
+
+            // No organization and no projects, we can create a new one
+            // using the initial setup config
+            this.logger.debug(
+                `Initial setup: Creating organization "${setup.organization.name}"`,
+            );
+
+            const organization = await this.organizationModel.create({
+                name: setup.organization.name,
+            });
+            const { organizationUuid } = organization;
```

Comment:
Why not destructure above? 
```suggestion
            const { organization } = await this.organizationModel.create({
                name: setup.organization.name,
            });
```